pallets/pool-registry/src/tests.rs

```diff
@@ -78,6 +80,7 @@ fn register_pool_and_set_metadata() {
 				currency,
 				max_reserve,
 				metadata.clone(),
+				()
```

Comment:
Updated trait and add pallet mock

From `pool-registry` perspective, we do not need to know more than the policy itself (whatever it is) reach the `update()` call.